sklearn/metrics/_classification.py

```diff
@@ -194,10 +194,11 @@ def _validate_multiclass_probabilistic_prediction(
     y_prob = check_array(
         y_prob, ensure_2d=False, dtype=[np.float64, np.float32, np.float16]
     )
+    xp, _ = get_namespace(y_true, y_prob)
 
-    if y_prob.max() > 1:
+    if xp.max(y_prob) > 1:
         raise ValueError(f"y_prob contains values greater than 1: {y_prob.max()}")
-    if y_prob.min() < 0:
+    if xp.min(y_prob) < 0:
```

Comment:
Not necessarily for this PR but this code seems to be repeated x4 in this module, maybe we could refactor it out?